lib/Client/Transports/TransportHTTPPROTO.php

```diff
@@ -73,6 +73,10 @@ public function flushReport($auth, $report) {
         @fwrite($fp, "POST /api/v2/reports HTTP/1.1\r\n");
         @fwrite($fp, $header . $content);
         @fflush($fp);
+        // Wait and read first line of the response e.g. (HTTP/1.1 2xx OK)
+        // otherwise the connection will close before the request is complete,
+        // leading to a context cancellation down stream.
+        fgets($fp);
```

Comment:
Looks good to me. For the additional viewing pleasure of others: https://www.php.net/manual/en/function.fsockopen.php#refsect1-function.fsockopen-examples

Reply:
Another thought: I'm not too familiar with this tracer -- but I imagine waiting for this response increases the time it takes to flush by a decent amount. Will this cause other issues?

Reply:
Hopefully not? For what it's worth it's moving away from a fire and forget to a fire and wait for the initial response. Reading the whole response body causes a good bit of added latency but I didn't notice any significant increase just reading the first line of the response.